### PR TITLE
In case a logger port is specified, redirect application logs to a socket.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

File: lib/ui/logger/system_logger.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+#ifndef FLUTTER_LIB_UI_LOGGER_SYSTEM_LOGGER_H_
+#define FLUTTER_LIB_UI_LOGGER_SYSTEM_LOGGER_H_
+
+#include <cstddef>
+#include <cstdint>
+
+namespace blink {
+
+void SytemLoggerLog(const uint8_t* message, size_t length);
```
Comment (jason-simmons) [inline lib/ui/logger/system_logger.h]: typo: "System"

File: lib/ui/logger/logger.cc [hunk 2]
```diff
@@ -0,0 +1,164 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/logger/logger.h"
```
Comment (abarth) [inline lib/ui/logger/logger.cc]: blank line after primary header

File: lib/ui/logger/logger.cc [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+#include <fcntl.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <unistd.h>
```
Comment (abarth) [inline lib/ui/logger/logger.cc]: blank line after system headers

File: lib/ui/logger/logger.cc [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    return false;
+  }
+  size_t sent = 0;
+  while (sent < allocation_size) {
+    int64_t sent_this_time = HANDLE_EINTR(::send(
+        client_socket_.get(), allocation + sent, allocation_size - sent, 0));
+    if (sent_this_time >= 0) {
+      sent += sent_this_time;
+    } else {
+      // This includes an EAGAIN in case the socket is full.
```
Comment (abarth) [inline lib/ui/logger/logger.cc]: You don't want to buffer or anything?  If the client doesn't pull from the socket fast enough, we'll lose data.

File: lib/ui/BUILD.gn [hunk 5]
```diff
@@ -12,6 +12,9 @@ source_set("ui") {
     "dart_runtime_hooks.h",
     "dart_ui.cc",
     "dart_ui.h",
+    "logger/logger.cc",
+    "logger/logger.h",
+    "logger/system_logger.h",
```
Comment (abarth) [inline lib/ui/BUILD.gn]: Seems like this code should be in a lower layer somewhere.  Doesn't seem specific to the UI library.

**Pull request closed without merging**